docs/proposals/configuration.md

```diff
@@ -0,0 +1,353 @@
+---
+title: Configuration Files Design
+
+authors:
+  - "@kadisi"
+  - "@fisherxu"
+  
+approvers:
+  - "@kevin-wangzefeng"
+  - "@sids-b"
+
+creation-date: 2019-10-02
+
+---
+
+* [Configuration Files Design](#configuration-files-design)
+   * [Abstract](#abstract)
+   * [Goals](#goals)
+   * [Proposal](#proposal)
+      * [cloudcore](#cloudcore)
+         * [config struct](#config-struct)
+         * [default config file](#default-config-file)
+      * [edgecore](#edgecore)
+         * [config struct](#config-struct-1)
+         * [default config file](#default-config-file-1)
+
+# Configuration Files Design
+
+
+## Abstract  
+
+Currently, cloudcore/edgecore/edgesite\`s configuration files are in the conf directory at the same level and have 3 configuration files, it is difficult to maintain and extend.
+
+Currently KubeEdge uses beehive's config package to analyse configuration files, but it has many drawbacks,when the program is running, it will output a lot of logs first.
+
+Is also no need to read configuration variables in multiple ways, such as commandline and env.
```

Comment:
commandline and env could be able to override the default config value, why this is needn't?

Reply:
We basically don't get variables from env, it is recommended to read the configuration file like k8s

Reply:
what's the result with some variables exposed in the env? ignore it?

Reply:
Set in the configuration file

Reply:
I guess we actually use this functionality in our e2e and perf tests where we override the default file conf using env

Reply:
@sids-b Can you provide an example?